src/constants/java/net/fabricmc/yarn/constants/CatTypes.java

```diff
@@ -0,0 +1,15 @@
+package net.fabricmc.yarn.constants;
+
+public final class CatTypes {
```

Comment:
These really needs a private constructor to prevent instantiation.

Reply:
If you take it seriously, all these classes are dangerous because they may be removed with mappings. Even if the constant values can be inlined by javac, you cannot prevent people from writing code like `synchronized (CatTypes.class)` even though you've privated its ctor.

Reply:
Perhaps have loom check and enforce the use of constant classes for the constants?

Reply:
preferably, these classes should be compile only like annotations are.

Reply:
Maybe it would be a good idea to have an annotation to signify that this class won't be available at runtime? Like we already have `@EnvType` which specifies which classes are available on the dedicated server.
This way, you could:
- Use the same system as already exists for `@EnvType` to refuse to load the class at runtime in dev, so such errors are caught before the mod is released to prod.
- Allow for tools such as mcdev to warn about such usages.

I don't think instantiation or synchronized would be a serious problem, but I can foresee some people trying to use reflection to list the fields of such classes.

Reply:
These classes are compile only in loom. If something is going to break it will most likely be noticed in dev quickly.